torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -679,16 +679,14 @@ bool ProcessGroupNCCL::CoalescedWorkNCCL::wait(
 static std::atomic<size_t> process_group_id = 0;
 
 ProcessGroupNCCL::ProcessGroupNCCL(
-    const c10::intrusive_ptr<Store>& store,
-    int rank,
-    int size,
+    DistributedBackendOptions& opts,
     c10::intrusive_ptr<Options> options)
-    : Backend(rank, size),
-      store_(store),
+    : Backend(opts.group_rank, opts.group_size),
+      store_(opts.store),
       options_(options),
```

Comment:
```suggestion
      options_(std::move(options)),
```
shouldn't this also be std::move? I am surprised clang-tidy did not catch this though.

Reply:
I don't think this should be an `std::move`.  The options object could be used inside this constructor too.